File: RUSH/Remove Nth Node From End of List.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} n
 * @return {ListNode}
 */
var removeNthFromEnd = function(head, n) {
    
    let slow = head ; 
    let fast = head ; 
    let prev = null ; 
    for (let i = 0 ; i < n ; i++){
        fast = fast.next ; 
    }

    while (fast){
        prev = slow 
        slow = slow.next 
        fast = fast.next 
    }

    if (prev !== null){
        prev.next = slow.next
        return head 
    }
    else {
        return head.next 
    }
};

File: RUSH/Remove Nth Node From End of List.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
from typing import Optional 

class Solution:

    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:

        prev = None 
        slow = head 
        fast = head 
        
        for i in range(n - 1): 
            fast = fast.next 
        
        while (fast.next): 
            prev = slow 
            slow = slow.next  
            fast = fast.next  
        
        if not prev is None : 
            
            prev.next = slow.next 
            return head 
            
        else : 
            
            return slow.next             